packages/dso-toolkit/gulp/build-svg-spritesheets.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-env node */
+
+import gulp from "gulp";
+import { basename } from "path";
+import { obj } from "list-stream";
+import { parse } from "css";
+import svgstore from "gulp-svgstore";
+import cheerio from "gulp-cheerio";
+import prettier from "gulp-prettier";
+import rename from "gulp-rename";
+
+import { sassTransformer } from "./transformers/sass.transformer.js";
+
+const iconsPath = "src/icons";
+const distPath = "dist";
+const outPutNameIcons = "dso-icons.svg";
+const outPutNameIconsWithVariants = "di.svg";
+const canvasSize = 32;
+const gutterSize = 10;
+
+async function parseStylesheets() {
+  return new Promise((resolve, reject) => {
+    gulp
+      .src(`${iconsPath}/*.scss`) // Read all SCSS files in the icons directory
+      .pipe(sassTransformer()) // Transform SCSS into plain CSS
+      .pipe(
+        obj((error, data) => {
+          if (error) {
+            return reject(error);
+          }
+
+          // Process each transformed SCSS file
+          const parsedStyles = data.map(({ relative, extname, contents }) => {
+            const id = basename(relative, extname); // Extract the file name without the extension
+            const style = contents.toString();
+            const ast = parse(style); // Parse the CSS into an abstract syntax tree (AST)
+
+            // Extract color variants for each CSS rule
+            const variants = ast.stylesheet.rules
+              .filter((rule) => rule.type === "rule") // Keep only CSS rules (ignore comments, etc.)
+              .flatMap((rule) => {
+                const colorDeclaration = rule.declarations.find((d) => d.property === "color");
+                if (!colorDeclaration) return [];
+
+                const color = colorDeclaration.value;
+                return rule.selectors
+                  .map((selector) => {
+                    if (selector === id) {
+                      return { selector: "default", color }; // Mark as default variant
+                    }
+
+                    if (selector.startsWith(`${id}:`)) {
+                      return { selector: selector.slice(id.length + 1), color }; // Extract the variant name
+                    }
+
+                    return null;
+                  })
+                  .filter(Boolean); // Remove null entries
+              });
+
+            return { id, variants };
+          });
+
+          resolve(parsedStyles);
+        }),
+      );
+  });
+}
+
+async function buildDiSpritesheet() {
+  const iconStylesheets = await parseStylesheets(); // Parse stylesheets within the function
+  return gulp
+    .src(`${iconsPath}/*.svg`)
+    .pipe(prettier({ plugins: ["@prettier/plugin-xml"] }))
+    .pipe(
+      svgstore({
+        inlineSvg: true,
+      }),
+    )
+    .pipe(
+      cheerio({
+        run($) {
+          if (iconStylesheets) {
```

Comment:
Kan deze wel eens `false` zijn?

En zo ja: Is dit dan de juiste plek? Ik zou dan willen dat de build in z'n geheel faalt ipv. doorgaat.

Reply:
Zeker, en nee. Ik heb doe nu na de parse functie een check en laat de build falen als er geen stylesheets geparsed zijn

Reply:
:+1: